API/Data/Repositories/MangaFileRepository.cs

```diff
@@ -32,4 +33,13 @@ public async Task<IList<MangaFile>> GetAllWithMissingExtension()
             .Where(f => string.IsNullOrEmpty(f.Extension))
             .ToListAsync();
     }
+
+    public async Task<MangaFile?> GetByKoreaderHash(string hash)
+    {
+        if (string.IsNullOrEmpty(hash)) return null;
+
+        return await _context.MangaFile
+            .FirstOrDefaultAsync(f => f.KoreaderHash != null &&
```

Comment:
Is there a reason we do a Like and not an equal? The Like is more expensive and there wont be any modifiers (%) for the like to perform on.

Reply:
I was going for a case insensitive search here originally is why.  However I realized I can accomplish the same thing by using .ToUpper() on the hash when searching since we store it as upper in the database.  Let me know if that works for you